doc/stdenv/stdenv.chapter.md

```diff
@@ -1057,7 +1057,55 @@ The `validatePkgConfig` hook validates all pkg-config (`.pc`) files in a package
 
 ### cmake
 
-Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake, however, if ninja is present, then the ninja generator will be used instead. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. By default, the cmake configurePhase can be thought of as:
+
+```sh
+mkdir build
+cd build
+cmake .. $cmakeFlags
+```
+
+#### Variables controlling CMake
+
+##### dontUseCmakeConfigure
+
+Avoid running `cmake` during the configure step altogether.
+
+##### cmakeFlags
+
+Flags passed to cmake. `cmakeFlags = "-DBUILD_SHARED_LIB=ON -DENABLE_LTO=ON";`
+
+##### cmakeFlagsArray
+
+Flag array passed to cmake. `cmakeFlagsArray = [ "-DBUILD_SHARED_LIB=ON" "-DENABLE_LTO=ON" ];`
```

Comment:
When grepping through the codebase `cmakeFlagsArray` seems to only be used in other contexts than described here such as `cmakeFlagsArray+=(-DCMAKE_EXE_LINKER_FLAGS="-ltbb -lcurl")` and never in the form as described here. Actually `cmakeFlags` is most often not a string, but a list of strings. It seems these docs don't match with how it's really used.

Reply:
it will still get stringified correctly:
```
toString ([ "a" "b"])
"a b"
```

just because my example isn't comprehensive, doesn't mean it's better than what existed... which was nothing